File: Submissions/94-binary-tree-inorder-traversal/binary-tree-inorder-traversal.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode() {}
 *     TreeNode(int val) { this.val = val; }
 *     TreeNode(int val, TreeNode left, TreeNode right) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */
class Solution {
    public List<Integer> inorderTraversal(TreeNode root) {
        List<Integer> L = new ArrayList<>();
        if (root == null)
            return L;
        if (root.left != null)
            L = inorderTraversal(root.left);
        L.add(root.val);
        if (root.right != null) 
            L.addAll(inorderTraversal(root.right));
        return L;

    }
}

File: Submissions/94-binary-tree-inorder-traversal/binary-tree-inorder-traversal.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution:
    def inorderTraversal(self, root: Optional[TreeNode]) -> List[int]:
        L = []
        if root == None:
            return []
        if root.left == None:
            L.append(root.val)
        else:
            L2 = self.inorderTraversal(root.left)
            L = L + L2
            L.append(root.val)
        if root.right != None:
            L3 = self.inorderTraversal(root.right)
            L = L + L3
        return L



        